src/eval/reduction.rs

```diff
@@ -936,9 +936,14 @@ fn apply_continuation<F: LurkField>(
                     Op1::Char => match result.tag {
                         ExprTag::Num | ExprTag::Char => {
                             let scalar_ptr = store
-                                .get_expr_hash(&result)
+                                .hash_expr(&result)
                                 .ok_or_else(|| store::Error("expr hash missing".into()))?;
-                            store.get_char_from_u32(scalar_ptr.value().to_u32_unchecked())
+                            // TODO: what should the behavior be on u32s which are not valid chars?
```

Comment:
Blocking: we should not pass through Rust characters for this exactly because of the issue noted. This is why there was a function called `get_char_from_u32`. All u32 values can be coerced to a valid Lurk char.